WalletWasabi.Fluent/ViewModels/Wallets/Buy/OrderViewModel.cs

```diff
@@ -87,10 +88,32 @@ public partial class OrderViewModel : ReactiveObject
 		// TODO: Remove this once we use newer version of DynamicData
 		HasUnreadMessagesObs.BindTo(this, x => x.HasUnreadMessages);
 
-		// Update file on disk
-		this.WhenAnyValue(x => x.HasUnreadMessages).Where(x => x == false).ToSignal()
-			.Merge(_messagesList.Connect().AutoRefresh(x => x.IsPaid).ToSignal())
-			.DoAsync(async _ => await UpdateConversationLocallyAsync(GetChatMessages(), _metaData, cancellationToken))
+		SaveConversationToFileWhenNeeded();
+	}
+
+	private IDisposable SaveConversationToFileWhenNeeded()
+	{
+		var flowCompleted = this.WhenAnyValue(x => x.WorkflowManager.CurrentWorkflow.IsCompleted)
+			.DistinctUntilChanged()
+			.Where(x => x).ToSignal()
+			.Do(_ => Logger.LogInfo($"{Title}: Flow completed"));
```

Comment:
I think this is not needed. because we call an API at the end of all Workflow, that not just saves the conversation to disc, but also sends it to SIB.